revocation/ocsp/ocsp.go

```diff
@@ -0,0 +1,195 @@
+// Package OCSP provides methods for checking the OCSP revocation status of a certificate chain
+// as well as errors related to these checks
+package ocsp
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"golang.org/x/crypto/ocsp"
+)
+
+// Logger is a simple logging interface based on what is logged in this package
+type Logger interface {
+	// Warnf logs a warn level message with format.
+	Warnf(format string, args ...interface{})
+	// Error logs an error level message.
+	Error(args ...interface{})
+}
+
+// NoOpLogger is a No-Op implementation of Logger
+type NoOpLogger struct{}
+
+// Warnf for NoOpLogger does nothing
+func (n *NoOpLogger) Warnf(format string, args ...interface{}) {}
+
+// Error for NoOpLogger does nothing
+func (n *NoOpLogger) Error(args ...interface{}) {}
```

Comment:
Yes that makes sense. I changed from printing to a logger based on your previous comment about not printing, but I see what you mean now.

I originally included the print statement for the sole purpose of addressing the id-pkix-ocsp-nocheck extension issue. In the future, we will need to use CRL if that extension isn't present. However, since we don't currently support CRL, there is a potential gap. Pritesh recommended logging this for the time being until we have CRL support, but I understand the hesitation to logging. I'll confirm with him about a potential alternative. Perhaps we just insert a TODO comment in the place of the print/log, then add the note to documentation?

Reply:
Update: I created a new error addressing the id-pkix-ocsp-nocheck extension. This will now be handled by the Verifier in notation-go, which will do the logging. All logging functionality for this package has been removed in the most recent commit